tiledb/sm/consolidator/fragment_consolidator.cc

```diff
@@ -362,7 +362,25 @@ Status FragmentConsolidator::consolidate_fragments(
   NDRange union_non_empty_domains;
   std::unordered_set<std::string> to_consolidate_set;
   for (auto& uri : fragment_uris) {
-    to_consolidate_set.emplace(uri);
+    if (uri.find('/') == std::string::npos) {
+      // The fragment URI is relative and does not contain the array URI.
+      to_consolidate_set.emplace(uri);
+    } else {
+      // The fragment URI is absolute and should contain the correct array URI.
+      URI fragment_uri(uri);
+      FragmentID frag_id(fragment_uri);
+
+      // Check for valid URI based on array format version.
+      auto fragments_dir = array_for_reads->array_directory().get_fragments_dir(
+          frag_id.array_format_version());
+      if (!fragment_uri.contains(fragments_dir)) {
```

Comment:
uri != fragments_dir.join_path(uri.last_path_part().c_str())

Reply:
Sorry to be nitpicky 🙃

Reply:
No worries :laughing: done